modernized Validation/DTRecHits

Dr15Jones opened the pull request:
DTHitQualityUtils was a class with only static members. This was
changed to be a namespace instead.
Converted global 'debug' variable to atomic.

Changed anonymous namespace global variable to be constexpr to avoid thread-safety issues.

Dr15Jones committed:
modernized DTHitQualityUtils

DTHitQualityUtils was a class with only static members. This was
changed to be a namespace instead.
Converted global 'debug' variable to atomic.

Dr15Jones commented:
The thread-safety problem was flagged by the static analyzer.

cmsbuild commented:
The code-checks are being triggered in jenkins.

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-23324/4820

cmsbuild commented:
A new Pull Request was created by  @Dr15Jones (Chris Jones) for master.

It involves the following packages:

Validation/DTRecHits


@vazzolini, @kmaeshima, @dmitrijus, @cmsbuild, @jfernan2, @vanbesien can you please review it and eventually sign? Thanks.
@davidlange6, @slava77, @fabiocos you are the release manager for this.

cms-bot commands are listed <a href="http://cms-sw.github.io/cms-bot-cmssw-cmds.html">here</a>

Dr15Jones committed:
Made anonymous namespace variable constexpr

The variable mirrorMinusWheels is only used in the file and is
only set as part of its declaration. This allows it to be constexpr
which also fixes the thread-safety warning.

cmsbuild commented:
The code-checks are being triggered in jenkins.

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-23324/4821

cmsbuild commented:
Pull request #23324 was updated. @vazzolini, @kmaeshima, @dmitrijus, @cmsbuild, @jfernan2, @vanbesien can you please check and sign again.

Dr15Jones commented:
please test

cmsbuild commented:
The tests are being triggered in jenkins.

jfernan2 commented:
@battibas I believe you are interested on this

cmsbuild commented:
+1
Tested at: a020504838f2741932c07811a856846a9e485b59
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-23324/28173/summary.html

The following merge commits were also included on top of IB + this PR after doing git cms-merge-topic: 
https://github.com/cms-sw/cmssw/commit/37116a8fb0afa281b196d689406c92d17fcd283e
https://github.com/cms-sw/cmssw/commit/11116700cd4d4f0d31add504017a2450e844c3f1
https://github.com/cms-sw/cmssw/commit/e901d7d688923e617d00adc99b3ee42cf65270f7
You can see more details here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-23324/28173/git-log-recent-commits
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-23324/28173/git-merge-result

cmsbuild commented:
Comparison job queued.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-23324/28173/summary.html

Comparison Summary:
 - No significant changes to the logs found
 - ROOTFileChecks: Some differences in event products or their sizes found
 - Reco comparison results: 2 differences found in the comparisons
 - DQMHistoTests: Total files compared: 31
 - DQMHistoTests: Total histograms compared: 2898511
 - DQMHistoTests: Total failures: 7700
 - DQMHistoTests: Total nulls: 3
 - DQMHistoTests: Total successes: 2890618
 - DQMHistoTests: Total skipped: 190
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: -0.168 KiB( 30 files compared)
 - DQMHistoSizes: changed ( 136.85,... ): -0.056 KiB       HLT/GeneralHLTOffline
 - Checked 128 log files, 14 edm output root files, 31 DQM output files

dmitrijus commented:
+1

cmsbuild commented:
This pull request is fully signed and it will be integrated in one of the next master IBs (tests are also fine). This pull request will now be reviewed by the release team before it's merged. @davidlange6, @slava77, @smuzaffar, @fabiocos (and backports should be raised in the release meeting by the corresponding L2)

fabiocos commented:
please test

just to cross check differences, which looks unrelated

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
+1
Tested at: a020504838f2741932c07811a856846a9e485b59
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-23324/28217/summary.html

cmsbuild commented:
Comparison job queued.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-23324/28217/summary.html

Comparison Summary:
 - No significant changes to the logs found
 - Reco comparison results: 0 differences found in the comparisons
 - DQMHistoTests: Total files compared: 31
 - DQMHistoTests: Total histograms compared: 2901706
 - DQMHistoTests: Total failures: 2
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 2901514
 - DQMHistoTests: Total skipped: 190
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 30 files compared)
 - Checked 128 log files, 14 edm output root files, 31 DQM output files

fabiocos commented:
+1
